### PR TITLE
Upgrade Kafka Clients API to 2.0.0

### PR DESCRIPTION
**Pull request by jeqo**
Changes: 
* Upgrading Kafka Clients versions to 2.0.0
* Upgrade Kafka JUnit dependency to 4.1.2

Solves: https://github.com/openzipkin/brave/issues/752

### CODE REVIEW

**Comment by jcchavezs** [comment]
I am curious about the tests here. Should we cover https://github.com/openzipkin/brave/pull/753/files#diff-1256abdd71e2b447c0bd97db22a69e72R52 or https://github.com/openzipkin/brave/pull/753/files#diff-1256abdd71e2b447c0bd97db22a69e72R259 at least?

**Comment by ImFlog** [comment]
You forgot to remove the `@Override` for new methods (to not break compatibility).
Sorry I said It was ok on the other thread but it is because I saw 
```
// Do not use @Override annotation to avoid compatibility issue version < 1.0
  public void subscribe(Pattern pattern) {
```

**Comment by jeqo** [comment]
@jcchavezs I've added a unit test case for `poll(Duration)` and add it to integration test also. WDYT?

File: instrumentation/kafka-clients/src/main/java/brave/kafka/clients/TracingConsumer.java [hunk 1]
```diff
@@ -49,12 +49,13 @@
     this.remoteServiceName = remoteServiceName;
   }
 
-  @Deprecated @Override public ConsumerRecords<K, V> poll(long timeout) {
-    return poll(Duration.ofMillis(timeout));
+  // Do not use @Override annotation to avoid compatibility issue version < 1.0
```
Comment (ImFlog) [inline instrumentation/kafka-clients/src/main/java/brave/kafka/clients/TracingConsumer.java]: `version < 2.0`
Also can you put this comment everywhere you removed the annotation ? May avoid some mistakes in the future ;)

**Comment by jeqo** [comment]
@ImFlog I have removed overrides.

File: instrumentation/kafka-clients/src/main/java/brave/kafka/clients/TracingConsumer.java (hunk 1 above)
Comment (jeqo) [reply instrumentation/kafka-clients/src/main/java/brave/kafka/clients/TracingConsumer.java]: Yeap

**Comment by ImFlog** [comment]
For what it's worth, LGTM :)

**Comment by adriancole** [comment]
So I will help on this, because I think it is important that we do a runtime check with kafka 1. It is somewhat tricky.

I will do this and rebase your branch over it (with maven invoker)

**Comment by adriancole** [comment]
OK I added the invoker test squashed and force pushed the branch. If
works and no other feedback we can merge

**Pull request merged**